### ISSUE TITLE
Concatenation problem

### ISSUE THREAD

**Issue by dupsys** [keras-team/keras#14248]
# Input char and embedding
inputs_char = Input(shape=(sequence_length,), name='input', dtype='int64') 
char_embedding  = embedding_layer_char(inputs_char)

# Input word and embedding
inputs_word = Input(shape=(sequence_length_word,), name='input', dtype='int64') 
word_embedding = embedding_layer_word(inputs_word)

# CNN model
filter_sizes = [1,2,3,5]
num_filters = 36

maxpool_pool = []
for i in range(len(filter_sizes)):
    conv = Conv1D(num_filters, kernel_size=(filter_sizes[i]),
                                 kernel_initializer='he_normal', activation='relu')(x)
    maxpool_pool.append(MaxPool1D(pool_size=(2))(conv))
    z = concatenate([max_time_pooling, word_embedding],axis=1)
    
#### Got the following error:

ValueError: A `Concatenate` layer requires inputs with matching shapes except for the concat axis. Got inputs shapes: [(None, 64, 128), (None, 140, 50)]

The input for char_embedding = [None,140,70] and word_embedding is [None, 140, 50]

Please, what do you advice i do? Thanks.

**Comment by tm-schwartz** [comment]
You are trying to concatenate inputs that do not match in shape. 
`z = concatenate([max_time_pooling, word_embedding],axis=1)`->`concatenate([(None, 64, 128), (None, 140, 50)],axis=1)`
For concatenate, the non-concat axis (axis 0 and 2) need to match, in your code axis 0 matches, but axis 2 of `max_time_pool` is **128** and axis 2 of `word_embedding` is **50**.  128 != 50 so you get ValueError. You will get a ValueError again if your batch sizes differ because then axis 0 of your layers will not match. 

So basically, get axis 2 of `max_time_pooling` and axis 2 of `word_embedding` to match, and make sure you are using the same batch size for your inputs.

**Comment by sushreebarsa** [comment]
@dupsys Moving this issue to closed status as there has been no recent activity.In case you still face the error please create a new issue,we will get you the right help.Thanks!
